### PR TITLE
Feature constraint previous move

### PR DESCRIPTION
**Pull request by paulSanchezMarmelab**
[feature_constraint_previous_move.webm](https://user-images.githubusercontent.com/130989826/233403515-10e09e70-f35a-4979-9da9-8be9a2c2e637.webm)

https://trello.com/c/7RIfw0X5/3-05-as-hayao-i-want-to-know-if-my-move-respect-constraints-set-up-by-the-previous-move

- [X] Implemented constraints
- [X] display unrespected move constraint inside error messages
- [X] made tests

### CODE REVIEW

File: src/move/move.spec.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    undefined,
+  ],
+  [
+    undefined,
+    undefined,
+    { symbol: "A", color: "magenta" },
+    { symbol: "A", color: "cyan" },
+    { symbol: "A", color: "green" },
+    { symbol: "D", color: "red" },
+    { symbol: "O", color: "green" },
```
Comment (adguernier) [inline src/move/move.spec.ts:30]: @paulSanchezMarmelab , a tile cannot has this value: O must be unique in the board with grey color

File: src/move/move.ts [hunk 2]
```diff
@@ -77,11 +78,34 @@ export const checkUserMove = (
...
+  ) {
+    let badMoveMessage = `🫠 Tile is not playable. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+
+    if (isColorConstraintUnrespected) {
+      badMoveMessage = `🫠 Tile has a different COLOR. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+    }
+
+    if (isSymbolConstraintUnrespected) {
+      badMoveMessage = `🫠 Tile has a different SYMBOL. Please player ${gameState.currentPlayer.toUpperCase()} choose a playable tile`;
+    }
```
Comment (slax57) [inline src/move/move.ts:103]: This method is already too long =>  All this new code should be moved to a new function

**Review by slax57 — CHANGES_REQUESTED** [review]
I however appreciate the tests in `src/move/move.spec.ts` which are nice and useful, well done! :slightly_smiling_face:

File: src/move/move.spec.ts (hunk 1 above)
Comment (paulSanchezMarmelab) [reply src/move/move.spec.ts]: Yes I saw that, and it broke mock board for tests. It should be fixed now.

File: src/move/move.ts (hunk 2 above)
Comment (paulSanchezMarmelab) [reply src/move/move.ts]: I refactored it.

File: src/tile/tile.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ export const colors = [
   "cyan",
   "magenta",
 ] as const;
-export const symbols = ["A", "B", "C", "D", "E", "F"] as const;
+export const symbols = ["A", "B", "C", "D", "E", "F", "O"] as const;
```
[the remark this answers is not in the document]
Comment (paulSanchezMarmelab) [reply src/tile/tile.ts]: This is fixed.

File: src/move/move.ts [hunk 4]
```diff
@@ -31,11 +28,56 @@ interface CheckedUserMove {
   allowedMove: boolean;
 }
 
+const checkMoveOnFirstTire = (
```
Comment (slax57) [inline src/move/move.ts:31]:
```suggestion
const checkMoveOnFirstTurn = (
```
?

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by slax57 — APPROVED** [review]

**Pull request merged**